### PR TITLE
Fix Examples

### LINKED ISSUE #132: Update android Examples
Examples don't work as needed, should be fixed.

### PR DESCRIPTION
**Pull request by korshaknn**
Please check 3 java activities:
ArActivity
SafetyActivity
ExternalCameraSource

VisionManager is fixed to prevent crash when we pass NOT a `Camera2VideoSourceImpl` as `videoSource` into `onCreate`

closes #132

### CODE REVIEW

File: Examples/src/main/java/com/mapbox/vision/examples/ArActivity.java [hunk 1]
```diff
@@ -72,93 +85,161 @@
...
+    }
 
-        // Route need to be reestablished if off route happens.
-        mapboxNavigation.addOffRouteListener(this);
-        mapboxNavigation.addProgressChangeListener(this);
+    private void startVisionManager() {
+        if (allPermissionsGranted() && !visionManagerWasInit) {
+            // Create and start VisionManager.
+            VisionManager.create();
+            VisionManager.setModelPerformanceConfig(new Merged(new On(ModelPerformanceMode.FIXED, ModelPerformanceRate.LOW)));
```
Comment (yunikkk) [inline Examples/src/main/java/com/mapbox/vision/examples/ArActivity.java]: Lets change to DYNAMIC performance config here and in other places

File: MapboxVision/src/main/java/com/mapbox/vision/VisionManager.kt [hunk 2]
```diff
@@ -186,8 +187,13 @@ object VisionManager : BaseVisionManager {
         sensorsManager = SensorsManager.Impl(application)
         locationEngine = LocationEngine.Impl(application)
 
-        val videoRecorder = SurfaceVideoRecorder.MediaCodecPersistentSurfaceImpl(application)
-        (videoSource as? Camera2VideoSourceImpl)?.setVideoRecorder(videoRecorder)
+        val videoRecorder = if (videoSource is Camera2VideoSourceImpl) {
```
Comment (yunikkk) [inline MapboxVision/src/main/java/com/mapbox/vision/VisionManager.kt]: Lets split this to separate commit? It's really bugfix and should be stated distinctively of the rest examples code changes.

Comment (korshaknn) [reply MapboxVision/src/main/java/com/mapbox/vision/VisionManager.kt]: done

File: Examples/src/main/java/com/mapbox/vision/examples/ArActivity.java (hunk 1 above)
Comment (korshaknn) [reply Examples/src/main/java/com/mapbox/vision/examples/ArActivity.java]: done

File: Examples/src/main/java/com/mapbox/vision/examples/ArActivity.java [hunk 3]
```diff
@@ -72,93 +85,161 @@
...
-                .setFastestInterval(1000)
-                .build();
-        locationCallback = new LocationEngineCallback<LocationEngineResult>() {
-            @Override
-            public void onSuccess(LocationEngineResult result) {}
-
-            @Override
-            public void onFailure(@NonNull Exception exception) {}
-        };
+    }
```
Comment (RingerJK) [inline Examples/src/main/java/com/mapbox/vision/examples/ArActivity.java]: `onCreate ` doesn't need here

Comment (korshaknn) [reply Examples/src/main/java/com/mapbox/vision/examples/ArActivity.java]: fixed

File: Examples/src/main/java/com/mapbox/vision/examples/BaseActivity.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.core.content.ContextCompat;
+
+import com.mapbox.vision.mobile.core.utils.SystemInfoUtils;
+import com.mapbox.vision.mobile.core.utils.snapdragon.SupportedSnapdragonBoards;
+import com.mapbox.vision.utils.VisionLogger;
+
+public abstract class BaseActivity extends AppCompatActivity {
+
+    private static final String PERMISSION_FOREGROUND_SERVICE = "android.permission.FOREGROUND_SERVICE";
```
Comment (RingerJK) [inline Examples/src/main/java/com/mapbox/vision/examples/BaseActivity.java]:
```suggestion
    private static final String PERMISSION_FOREGROUND_SERVICE = Manifest.permission.html.FOREGROUND_SERVICE;
```

File: Examples/src/main/java/com/mapbox/vision/examples/BaseActivity.java [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        String board = SystemInfoUtils.INSTANCE.getSnpeSupportedBoard();
+        if (!SupportedSnapdragonBoards.isBoardSupported(board)) {
+            Spanned text =
+                    Html.fromHtml("The device is not supported, you need <b>Snapdragon-powered</b> device with <b>OpenCL</b> support, more details at <b>https://www.mapbox.com/android-docs/vision/overview/</b>");
+            Toast.makeText(this, text, Toast.LENGTH_LONG).show();
+            VisionLogger.Companion.e(
```
Comment (RingerJK) [inline Examples/src/main/java/com/mapbox/vision/examples/BaseActivity.java]: hmmm, is it work?

File: Examples/src/main/java/com/mapbox/vision/examples/ExternalCameraSource.java [hunk 6]
```diff
@@ -68,60 +69,93 @@ public void detach() {
...
+        }
 
         @Override
-        public void onUpdateCompleted() {}
+        public void onUpdateCompleted() {
+        }
     };
 
     @Override
     public void onCreate(Bundle savedInstanceState) {
```
Comment (RingerJK) [inline Examples/src/main/java/com/mapbox/vision/examples/ExternalCameraSource.java]: unnecessary `onCreate`

File: Examples/src/main/java/com/mapbox/vision/examples/BaseActivity.java (hunk 4 above)
Comment (korshaknn) [reply Examples/src/main/java/com/mapbox/vision/examples/BaseActivity.java]: such field requires API level 28. current min is 23

File: Examples/src/main/java/com/mapbox/vision/examples/ExternalCameraSource.java (hunk 6 above)
Comment (korshaknn) [reply Examples/src/main/java/com/mapbox/vision/examples/ExternalCameraSource.java]: outdated

File: Examples/src/main/java/com/mapbox/vision/examples/BaseActivity.java (hunk 5 above)
Comment (korshaknn) [reply Examples/src/main/java/com/mapbox/vision/examples/BaseActivity.java]: why it shouldn't work?

File: MapboxVision/src/main/java/com/mapbox/vision/video/videosource/camera/SurfaceVideoRecorder.kt [hunk 7]
```diff
@@ -15,6 +15,25 @@ internal interface VideoRecorder {
     fun startRecording(path: String)
     fun stopRecording()
     fun release()
+
+    class EmptyImpl : VideoRecorder {
```
Comment (RingerJK) [inline MapboxVision/src/main/java/com/mapbox/vision/video/videosource/camera/SurfaceVideoRecorder.kt]: we already have `DummyImageSaver` and `DummyEventManager`. Lets keep the same style. `DummyVideoRecorder`

Comment (korshaknn) [reply MapboxVision/src/main/java/com/mapbox/vision/video/videosource/camera/SurfaceVideoRecorder.kt]: done

File: Examples/src/main/java/com/mapbox/vision/examples/BaseActivity.java (hunk 5 above)
Comment (korshaknn) [reply Examples/src/main/java/com/mapbox/vision/examples/BaseActivity.java]: discussed, fixed

**Comment by korshaknn** [comment]
@RingerJK ptal

File: Examples/src/main/java/com/mapbox/vision/examples/BaseActivity.java [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.core.content.ContextCompat;
+
+import com.mapbox.vision.mobile.core.utils.SystemInfoUtils;
+import com.mapbox.vision.mobile.core.utils.snapdragon.SupportedSnapdragonBoards;
+import com.mapbox.vision.utils.VisionLogger;
+
+public abstract class BaseActivity extends AppCompatActivity {
+
+    private static final String PERMISSION_FOREGROUND_SERVICE = "android.permission.FOREGROUND_SERVICE";
```
Comment (RingerJK) [inline Examples/src/main/java/com/mapbox/vision/examples/BaseActivity.java]: why you decided don't change it

Comment (korshaknn) [reply Examples/src/main/java/com/mapbox/vision/examples/BaseActivity.java]: see the comment, it requires API 28

**Comment by korshaknn** [comment]
@RingerJK fixed kotlin activities, ptal

**Pull request merged**